framework/math/ParallelVector/ParallelVector.h

```diff
@@ -0,0 +1,164 @@
+   * Set the entries of the locally owned portion of the parallel vector
+   * to the given value.
+   */
+  virtual void Set(double value) = 0;
+
+  /**
+   * Set the entries of the locally owned portion of the parallel vector
+   * to the given STL vector.
+   */
+  virtual void Set(const std::vector<double>& local_vector) = 0;
+
+  /**
+   * Define a set or add operation for the given global id-value pair
+   *
+   * This routine adds the global id-value pair to the set operation cache,
+   * which upon execution of Assemble, communicates the operations to the
+   * appropriate process.
+   */
+  virtual void SetValue(int64_t global_id, double value, VecOpType op_type) = 0;
+
+  /**
+   * Group multiple operations into a single call.
+   *
+   * This routine goes through the given global id-value pairs and calls
+   * SetValue for each.
+   */
+  virtual void SetValues(const std::vector<int64_t>& global_ids,
+                         const std::vector<double>& values,
+                         VecOpType op_type) = 0;
+
+  /**In place adding of vectors. The sizes must be compatible.*/
+  virtual void operator+=(const ParallelVector& y) = 0;
+
+  /**Adds a vector multiplied by scalar a. Optimized for a=1.0 and -1.0*/
+  virtual void PlusAY(double a, const ParallelVector& y) = 0;
+
+  /**Sets the local values of one vector equal to another. The sizes must be
+   * compatible.*/
+  virtual void CopyValues(const ParallelVector& y) = 0;
+
+  /**Copies a contiguous block of local data from the source vector to the
+   * current vector. The blocks are specified with offset values and num_values.
+   * */
+  virtual void BlockCopyLocalValues(const ParallelVector& y,
+                                    int64_t y_offset,
+                                    int64_t local_offset,
+                                    int64_t num_values) = 0;
+
+  /**Copies a contiguous block of local data from the source STL vector to the
+   * current vector. The local block offset is specified with the local-offset
+   * value. The input STL vector must have num_values entries.
+   * */
+  virtual void BlockCopyLocalValues(const std::vector<double>& y,
+                                    int64_t local_offset,
+                                    int64_t num_values) = 0;
+
+  /**Returns the specified norm of the vector.*/
+  virtual double ComputeNorm(chi_math::NormType norm_type) const = 0;
+
+  /**
+   * Communicate all operations stored within the operation cache to the
+   * corresponding processes that own the respective global indices, and
+   * apply the operations.
+   *
+   * This routine clears the respective operation cache once completed.
+   */
+  virtual void Assemble() = 0;
+
+  /// Print the local vectors to stings.
+  virtual std::string PrintStr() const = 0;
+
+  /**
+   * Communicate the current ghost entries, if applicable, to all other
+   * processes to update the locally stored ghost data.
+   */
+  virtual void CommunicateGhostEntries(){};
```

Comment:
```suggestion
  /**
   * Communicate the current ghost entries, if applicable, to all other
   * processes to update the locally stored ghost data.
   */
  virtual void CommunicateGhostEntries() {};
```
We've come full circle from drawing a line in the sand with using the word "ghost" in a non-ghost vector class

Reply:
I've already messaged you in private and said I conceded. If you want a public concession then here it is. Congrats Zach, our disagreement tilted in your favor. Now let's do better next time and not put things like this in a review.